docs/jmespath-grammar.txt

```diff
@@ -0,0 +1,133 @@
+expr      = ws ( terminal / non-terminal ) ws
+root-expr = non-test / not / comparison
+non-test  = subexpr / index / flatten / wildcard-index / filter /
```

Comment:
This is to break the grammar into smaller reusable parts, but has the side effect of an awkward node. These nodes can be used in the filter-condition, not and comparator rules without ambiguity. Maybe there's a better name or perhaps duplication is better.